packages/js/src/ws/socket.ts

```diff
@@ -188,10 +189,33 @@ export class Socket extends BaseModule {
 
     this.callWithSession(async () => {
       this.#initializeSocket().catch((error) => {
+        // eslint-disable-next-line no-console
         console.error(error);
       });
 
       return {};
+    }).catch((error) => {
+      // eslint-disable-next-line no-console
+      console.error(error);
+    });
```

Comment:
all promises should have either `catch` or `then`

Reply:
How about returning the promise to the customer and let them do the error handling? This console.error is very arbitrary and prevents them from logging such errors to their error reporter.